rust_cli/src/cli.rs

```diff
@@ -0,0 +1,180 @@
+use clap::{App, Arg, SubCommand};
+
+pub fn get_clap_app() -> App<'static, 'static> {
+    App::new("rust-cli")
+        .about("A Solana CLI application")
+        .version("0.1.0")
+        
+        .subcommand(
+            SubCommand::with_name("idl")
+                .about("IDL management commands")
+                .subcommand(
+                    SubCommand::with_name("upload")
+                        .about("Upload IDL from JSON file")
+                        .arg(
+                            Arg::with_name("idl-path")
+                                .help("Path to the IDL JSON file")
+                                .required(true)
+                                .index(1)
+                        )
+                        .arg(
+                            Arg::with_name("program-id")
+                                .short("pid")
```

Comment:
does this short command actually work? Im afraid the p might conflict with the p from the fees?

Reply:
Since program id is mandatory it does not need a flag at all. Removed the short p - so p for priority fees should not be in question.